chore: L1 confirmation checker (#87)
OffchainLabs/arbitrum-tutorials: packages/l1-confirmation-checker/README.md

# l1 confirmation checker Tutorial

`l1 confirmation checker` is a simple demo of Arbitrum's transaction finality checker (used to check if transaction submitted to l1 or not).

> It calls precompile `NodeInterface` to check it related l1 transaction.

Reviewer: ```suggestion
It calls precompile `NodeInterface` to find information about the L1 transactions that posted an L2 transaction in a batch.
```

> # l1 confirmation checker Tutorial

Reviewer 2: ```suggestion
# L1 Confirmation Checker Tutorial
```

`l1 confirmation checker` is a simple demo of Arbitrum's transaction finality checker (used to check if transaction submitted to l1 or not).

> It calls precompile `NodeInterface` to find information about an L1 transactions that posted the L2 transaction in a batch.

Reviewer 2: ```suggestion
It calls precompile `NodeInterface` to find information about an L1 transaction that posted the L2 transaction in a batch.
```

> It has 2 functions, first is `checkConfirmation` which you can use to check if your l2 transaction is confirmed on L1 or not, second is `findSubmissionTx` which can be used when your L2 transaction is confirmed on L1, you can call it to find what the L1 submission transaction is.

Reviewer: A suggestion for clarification, feel free to edit/

```suggestion
It has 2 functions; both function will show you whether you L2 transaction has been posted in an L1 batch. 
The first, `checkConfirmation`, will output the number of L1 block confirmations the L1 batch-posting transaction has.
 The second is `findSubmissionTx`, will output the L1 batch-posting transaction hash.
```

See [./exec.js](./scripts/exec.js) for inline explanation.

### Run Demo:

> check if tx recorded to l1 or not:

Reviewer 2: ```suggestion
Check if tx recorded in l1 or not:
```

```
yarn ts-node scripts/exec.ts --action checkConfirmation --txHash {YOUR_TX_HASH}
```

> get submissiontx by a given l2 transaction status:

Reviewer 2: ```suggestion
Get submissiontx by a given l2 transaction status:
```
